docs/Configuration.md

```diff
@@ -37,6 +37,7 @@ You can add these configurations into spark-defaults.conf to enable or disable t
 | spark.gluten.sql.columnar.broadcastJoin                 | Enable or Disable Columnar BroadcastHashJoin, default is true                                                                                                                                                                                                                                                                                                                                                                                        | true                                                 |
 | spark.gluten.sql.columnar.shuffle.codec                 | Set up the codec to be used for Columnar Shuffle. If this configuration is not set, will check the value of spark.io.compression.codec. By default, Gluten use software compression. Valid options for software compression are lz4, zstd. Valid options for QAT and IAA is gzip.                                                                                                                                                                    | lz4                                                  |
 | spark.gluten.sql.columnar.shuffle.codecBackend          | Enable using hardware accelerators for shuffle de/compression. Valid options are QAT and IAA.                                                                                                                                                                                                                                                                                                                                                        |                                                      |
+| spark.gluten.sql.columnar.shuffle.compressionMode          | Setting different compression mode in shuffle, Valid options are buffer and rowvector, buffer means compress each buffer of RowVector to pre allocated big buffer, rowvector means to copy the buffers of RowVector to a big buffer, and then compress the buffer.                                                                                                                                                                                                                                                                                                                                                        | buffer                                                 |
```

Comment:
Probably better explanation:

`buffer` option compress each buffer of RowVector individually into one pre-allocated large buffer. `rowvector` option first copies each buffer of RowVector to a large buffer and then compress the entire buffer in one go.